modules/qc/1.0/qc.smk

```diff
@@ -0,0 +1,434 @@
+    params:
+        opts = CFG["options"]["QualityScoreDistribution"],
+        jvmheap = lambda wildcards, resources: int(resources.mem_mb * 0.8)
+    conda:
+        CFG["conda_envs"]["gatkR"]
+    threads:
+        CFG["threads"]["QualityScoreDistribution"]
+    resources:
+        **CFG["resources"]["QualityScoreDistribution"]
+    shell:
+        op.as_one_line("""
+        echo "running {rule} for {wildcards.sample_id} on $(hostname) at $(date)" >> {log.stdout};
+        gatk
+        --java-options "-Xmx{params.jvmheap}m -XX:ConcGCThreads=1"
+        QualityScoreDistribution
+        {params.opts}
+        -I {input.bam}
+        -O {output.gatk_basequal}
+        -CHART {output.gatk_basequal_chart}
+        -R {input.fasta}
+        >> {log.stdout}
+        2>> {log.stderr} &&
+        echo "DONE {rule} for {wildcards.sample_id} on $(hostname) at $(date)" >> {log.stdout};
+        """)
+
+
+# Collect GATK QC metrics (separately for genomes/exomes)
+rule _qc_gatk_wgs:
+    input:
+        bam = str(rules._qc_input_bam.output.bam),
+        fasta = str(rules._qc_input_references.output.genome_fa),
+        samtools_stats = str(rules._qc_samtools_stat.output.samtools_stat)
+    output:
+        gatk_wgs = CFG["dirs"]["gatk"] + "CollectMetrics/{seq_type}--{genome_build}/{sample_id}.{genome_build}.CollectWgsMetrics.txt"
+    log:
+        stdout = CFG["logs"]["gatk"] + "CollectMetrics/{seq_type}--{genome_build}/{sample_id}.CollectWgsMetrics.stdout.log",
+        stderr = CFG["logs"]["gatk"] + "CollectMetrics/{seq_type}--{genome_build}/{sample_id}.CollectWgsMetrics.stderr.log"
+    wildcard_constraints:
+        seq_type = "genome"
+    params:
+        opts = CFG["options"]["CollectWgsMetrics"],
+        jvmheap = lambda wildcards, resources: int(resources.mem_mb * 0.8)
+    conda:
+        CFG["conda_envs"]["gatkR"]
+    threads:
+        CFG["threads"]["CollectMetrics"]
+    resources:
+        **CFG["resources"]["CollectMetrics"]
+    shell:
+        op.as_one_line("""
+        echo "running {rule} for {wildcards.sample_id} on $(hostname) at $(date)" >> {log.stdout};
+        gatk
+        --java-options "-Xmx{params.jvmheap}m -XX:ConcGCThreads=1"
+        CollectWgsMetrics
+        {params.opts}
+        -I {input.bam}
+        -O {output.gatk_wgs}
+        -R {input.fasta}
+        --READ_LENGTH $(grep ^SN {input.samtools_stats} | cut -f 2- | grep "average length:" | cut -f 2)
+        >> {log.stdout}
+        2>> {log.stderr} &&
+        echo "DONE {rule} for {wildcards.sample_id} on $(hostname) at $(date)" >> {log.stdout};
+        """)
+
+
+def _qc_get_baits(wildcards):
+    CFG = config["lcr-modules"]["qc"]
+    # simplify config options by re-using the capture space bertween e.g grch37 and hs37d5
+    if "38" in str(wildcards.genome_build):
+        this_genome_build = "hg38"
+    else:
+        this_genome_build = "grch37"
+
+    if "baits_regions" in  CFG:
```

Comment:
Any interest in making in an option to specify separate baits and regions files? Picard does output different metrics for baits vs. regions if different files are given.

Reply:
certainly can be added! Not supported in current implementation but definitely an option for the future update! I envision this to be also an option for the samtools stats, it is possible to restrict to target regions there